submissions/SergSenras/js-dom/script.js

```diff
@@ -0,0 +1,93 @@
+}
+
+function clear(){
+  const pageContent = document.getElementsByTagName('main')[0];
+  pageContent.innerHTML = '';
+}
+
+function displayContent(key){
+  clear();
+  const container = document.getElementsByTagName('main')[0];
+  const fragment = document.createDocumentFragment();
+  
+  contentActivators = {
+    'nav__main': function(){
+      clear();
+      const wrapper = document.createElement('div');
+      wrapper.className = 'item__box';
+      const textEl = document.createElement('p');
+      const imageEl = document.createElement('img');
+      imageEl.className = 'img__common';
+      textEl.innerText = main.content;
+      imageEl.src = main.image;
+      wrapper.appendChild(imageEl);
+      wrapper.appendChild(textEl);
+      return wrapper;
+    },
+    'nav__plot': function(){
+      clear();
+      const wrapper = document.createElement('div');
+      wrapper.className = 'item__box';
+      const textEl = document.createElement('p');
+      textEl.innerText = plot.content;
+      wrapper.appendChild(textEl);
+  
+      return wrapper;
+    },
+    'nav__towns': function(){
+      for(var i in towns){
+        const wrapper = document.createElement('div');
+        wrapper.className = 'item__box';
+        const headerEl = document.createElement('h2');
+        const textEl = document.createElement('p');
+        const imageEl = document.createElement('img');
+        imageEl.className = 'img__town img__common';
+        headerEl.innerText = towns[i].name;
+        textEl.innerText = towns[i].content;
+        imageEl.src = towns[i].image;
+        wrapper.appendChild(headerEl);
+        wrapper.appendChild(imageEl);
+        wrapper.appendChild(textEl);
+        fragment.appendChild(wrapper);
```

Comment:
Different render logic for 'nav__towns'. Looks confusing.

Besides, these methods (content activators) are all imperative, while better code style is more declarative. Once you switch to more declarative style you will suffice with a single function to render any of your content sources.

Read on declarative vs imperative, try, ask for clarifications if needed.

Keep in mind again, that the code is for other people. Someone should be able to maintain the code you've written for them.

Reply:
```suggestion
function createMarkup(obj){
  var wrapper = document.createElement('div');
  for(var i = 0; i < obj.length; i++){
    var contentBox = document.createElement('div');
    var elem;
    for(var prop in obj[i]){
      contentBox.className = 'item__box';
      if(prop == 'name') {
        elem = document.createElement('h2');
        elem.innerText = obj[i][prop];
      } else if(prop == 'image') {
        elem = document.createElement('img');
        elem.className = 'img__common';
        elem.src = obj[i][prop];
      } else if(prop == 'content') {
        elem = document.createElement('p');
        elem.innerText = obj[i][prop];
      }
      contentBox.appendChild(elem);
    }
    wrapper.appendChild(contentBox);
  }
  return wrapper;
}
```
Tried to make it in on function but not sure about imperative